dart/lib/src/sentry_envelope.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:convert';
+
+import 'sentry_envelope_header.dart';
+import 'sentry_envelope_item.dart';
+import 'protocol/sentry_event.dart';
+import 'protocol/sdk_version.dart';
+
+/// Class representation of `Envelope` file.
+class SentryEnvelope {
+  SentryEnvelope(this.header, this.items);
+
+  /// Header descriping envelope content.
+  final SentryEnvelopeHeader header;
+
+  /// All items contained in the envelope.
+  final List<SentryEnvelopeItem> items;
+
+  /// Create an `SentryEnvelope` with containing one `SentryEnvelopeItem` which holds the `SentyEvent` data.
+  factory SentryEnvelope.fromEvent(SentryEvent event, SdkVersion sdkVersion) {
+    return SentryEnvelope(SentryEnvelopeHeader(event.eventId, sdkVersion),
+        [SentryEnvelopeItem.fromEvent(event)]);
+  }
+
+  /// Stream binary data representation of `Envelope` file encoded in utf8.
+  Stream<List<int>> envelopeStream() async* {
+    yield utf8.encode(jsonEncode(header.toJson()));
+    final newLineData = utf8.encode('\n');
```

Comment:
Are we adding 2 `\n` on the first iteration?
I believe this line can be removed, no?
```suggestion
```

If there's no item it'll be an envelope with a header only, without the delimiter but I believe that is even accepted by Relay even though it's useless at this point.

Reply:
iirc the endpoint will return an error written "empty envelope", in such case, we'd not even send a request, but rather dropping it.

Reply:
We only append the `newLineData` in the loop, before every item. If there are no items, it's only the header item without a new line, which should be correct? Or am i missing something here?

Reply:
this line just declares the `newLineData`, appending it is only on L29 as @denrase said